internal/util-interface/src/main/java/xsbti/Problem.java

```diff
@@ -26,4 +28,24 @@ public interface Problem {
   default Optional<String> rendered() {
     return Optional.empty();
   }
+
+  /**
+   * The unique code attached to the diagnostic being reported.
+   *
+   * <p>NOTE: To avoid breaking compatibility we provide a default to account for older Scala
+   * version that do not have codes.
+   */
+  default Optional<DiagnosticCode> diagnosticCode() {
```

Comment:
Actually this issue seems to be fixed already in zinc https://github.com/sbt/zinc/pull/1082

Reply:
Sort of, the fix means that errors coming from mapped position bypass the renderer, but that means they get rendered by sbt which lacks the error code and displays error differently from scala 3, so it makes things pretty inconsistent.

Reply:
huh, then they miss out on the `explain` as well even don't they? The one part that I'm not sure yet is about the `explain`. Since the `explain` is attached to the message, not the actual code, and at times the message/explain is dynamically put together for different cases that re-use the same code. So it goes away a _bit_ from the actual meaning of the description of the code, so I'm not sure we can stick the explain text in there. wdyt? I mean if we can, then sure then it'd be up to the tool to take the code, the description (explain), and any related information and put together the rendered output.

Reply:
I don't see a problem with putting the explain part in there.